src/main/java/ch/njol/skript/expressions/ExprCommandInfo.java

```diff
@@ -43,108 +47,105 @@
+			"[the] label [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] label",
+			"[the] usage [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] usage",
+			"[(all|the|all [of] the)] aliases [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] aliases",
+			"[the] permission [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] permission",
+			"[the] permission message [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] permission message",
+			"[the] plugin [owner] [of [[the] command[s] %-strings%]]", "command[s] %strings%'[s] plugin [owner]");
 	}
 
-	@SuppressWarnings("null")
-	InfoType type;
-	@SuppressWarnings("null")
-	Expression<String> commandName;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private InfoType type;
+
+	@Nullable
+	private Expression<String> commandName;
 
 	@Override
+	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		commandName = (Expression<String>) exprs[0];
+		if (commandName == null && !getParser().isCurrentEvent(ScriptCommandEvent.class))
+			return false;
 		type = InfoType.values()[Math.floorDiv(matchedPattern, 2)];
 		return true;
 	}
 
 	@Nullable
 	@Override
-	@SuppressWarnings("null")
-	protected String[] get(Event e) {
-		CommandMap map = Commands.getCommandMap();
-		Command[] commands = this.commandName.stream(e).map(map::getCommand).filter(Objects::nonNull).toArray(Command[]::new);
-		ArrayList<String> result = new ArrayList<>();
-		switch (type) {
-			case NAME:
-				for (Command command : commands)
-					result.add(command.getName());
-				break;
-			case DESCRIPTION:
-				for (Command command : commands)
-					result.add(command.getDescription());
-				break;
-			case LABEL:
-				for (Command command : commands)
-					result.add(command.getLabel());
-				break;
-			case USAGE:
-				for (Command command : commands)
-					result.add(command.getUsage());
-				break;
-			case ALIASES:
-				for (Command command : commands)
-					result.addAll(command.getAliases());
-				break;
-			case PERMISSION:
-				for (Command command : commands)
-					result.add(command.getPermission());
-				break;
-			case PERMISSION_MESSAGE:
-				for (Command command : commands)
-					result.add(command.getPermissionMessage());
-				break;
-			case PLUGIN:
-				for (Command command : commands) {
-					if (command instanceof PluginCommand) {
-						result.add(((PluginCommand) command).getPlugin().getName());
-					} else if (command instanceof BukkitCommand) {
-						result.add("Bukkit");
-					} else if (command.getClass().getPackage().getName().startsWith("org.spigot")) {
-						result.add("Spigot");
-					} else if (command.getClass().getPackage().getName().startsWith("com.destroystokyo.paper")) {
-						result.add("Paper");
-					}
-				}
-				break;
+	protected String[] get(Event event) {
+		Command[] commands;
+		if (commandName == null) {
```

Comment:
can we move this `commands` computation logic to it's own private method? that way it'll be easier to follow